assets/scripts/app/flag_utils.js

```diff
@@ -39,22 +46,60 @@ function initRedrawPaletteUpdateListener () {
   return observeStore(select, onChange)
 }
 
-export function receiveUserFlags (flags) {
+/**
+ * Creates and returns an Object in shape of { source, flags: [ { key: value } ], priority }
+ *
+ * @param {Object} flags
+ * @param {String} source
+ */
+export function generateFlagOverrides (flags, source) {
   if (!flags) return
 
-  const sessionFlags = (window.localStorage.flags) ? JSON.parse(window.localStorage.flags) : {}
-  // Convert to array
-  const array = Object.entries(flags)
-  // Filter flags not in session flags
-  const filter = array.filter((item) => !(item[0] in sessionFlags))
-  // Convert back to object
-
-  return filter.reduce((obj, item) => {
-    obj[item[0]] = {
-      value: item[1],
-      source: 'user'
+  const flagsOverrides = {
+    source,
+    flags: [],
+    priority: PRIORITY_LEVELS[source]
+  }
+
+  return Object.entries(flags).reduce((obj, item) => {
+    const [key, value] = item
+    const flag = {
+      flag: key,
+      value
     }
 
+    flagsOverrides.flags.push(flag)
     return obj
-  }, {})
+  }, flagsOverrides)
+}
+
+/**
+ * Dispatches action to apply flag overrides to Redux flag state
+ *
+ * @param {Array} flagOverrides
+ * @param {Object} defaultFlags
+ */
+export function applyFlagOverrides (flagOverrides, defaultFlags) {
```

Comment:
We might want to redefine the function signature with the arguments swapped, like so:

```js
function applyFlagOverrides (defaultFlags, flagOverrides) {}
```

So that the object to be overridden appears first, then overrides come second. This matches the behavior of `Object.assign(target, overrides)`. One thing that this allows is any number of overrides to exist as arguments, because this also works: `Object.assign(target, override1, override2, override3, ...)` To make this work, you would "collect" all of the later arguments under one thing, with "rest" operator, like so:

```js
function applyFlagOverrides (defaultFlags, ...flagOverrides) {}
```

That way, in `authentication.js`, when you write this:

```js
applyFlagOverrides([sessionOverrides, userOverrides], store.getState().flags)
```

You could accomplish the same thing like this:

```js
applyFlagOverrides(store.getState().flags, userOverrides, sessionOverrides)
```

Now each override is its own argument, and you could provide any number of arguments. You no longer have to worry about putting them in an array yourself, as the rest operator does it for you automatically. (I've also swapped the arguments in order of priority, but the results should be the same.)